var/spack/repos/builtin/packages/rocrand/package.py

```diff
@@ -120,13 +121,38 @@ class Rocrand(CMakePackage):
     # same directory.
     patch("hiprand_prefer_samedir_rocrand.patch", working_dir="hiprand", when="@5.2.0:")
 
+    # Add hiprand sources thru the below
+    for d_version, d_commit in [
+        ("5.3.0", "125d691d3bcc6de5f5d63cf5f5a993c636251208"),
+        ("5.2.3", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.2.1", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.2.0", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.1.3", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
+        ("5.1.0", "7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b"),
```

Comment:
How did you decide on these versions? I'd have expected them to match the commit id listed by the hipRAND submodule in rocRAND.

```
5.3.0: 12e2f070337945318295c330bf69c6c060928b9e
5.2.3: 12e2f070337945318295c330bf69c6c060928b9e
5.2.1: 12e2f070337945318295c330bf69c6c060928b9e
5.2.0: 12e2f070337945318295c330bf69c6c060928b9e
5.1.3: 20ac3db9d7462c15a3e96a6f0507cd5f2ee089c4
5.1.3: 20ac3db9d7462c15a3e96a6f0507cd5f2ee089c4
```

Reply:
Hi @cgmb ,
These Commit IDs are from hiprand repo w.r.t the time stamps of 5.3.0 and 5.2.3 release tags of rocrand.
Commit IDs [125d691d3bcc6de5f5d63cf5f5a993c636251208 ](https://github.com/ROCmSoftwarePlatform/hipRAND/commit/125d691d3bcc6de5f5d63cf5f5a993c636251208) from the hipRAND repo (https://github.com/ROCmSoftwarePlatform/hipRAND) was taken based on the timestamp of release tag of 5.3.0.
Prior to the 5.3.0 related changes older release had not conflicts with the commit ID [7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b](https://github.com/ROCmSoftwarePlatform/hipRAND/commit/7190cc2199d8d97dbb6ce3fd179f1a62ad8fb13b).
Hence it is used for 5.1.0 through 5.2.3.

Reply:
commit ids were picked based on the rocm releases(5.2.0, 5.2.3,5.3.0 ) tag's time stamp of the rocrand . I tried to match the respective commit id for the hiprand.

Reply:
I don't think the timestamp is the correct way to choose the corresponding hipRAND version. Releases are made from a feature branch, so the development branch may have added commits intended for the next release at the time the previous release is tagged.